src/wp-includes/class-wp-user.php

```diff
@@ -876,9 +899,7 @@ public function for_site( $site_id = '' ) {
 
 		$this->cap_key = $wpdb->get_blog_prefix( $this->site_id ) . 'capabilities';
 
-		$this->caps = $this->get_caps_data();
-
-		$this->get_role_caps();
```

Comment:
I think removing this without replacing it with some other mechanism to "reset" the flag that capability and role data are loaded will lead to problems. For example, if capabilities and roles were already loaded and then someone calls `WP_User::for_site()` with another site ID, the `WP_User` object would still have the old capabilities and roles, not the ones for the site requested.

Reply:
Indeed, this appears to be the cause of the test failures for MS.

Reply:
Fixed.